File: app/src/main/java/com/dynamic/dobler/elephantmath/viewModel/ChallengeViewModel.java
package com.dynamic.dobler.elephantmath.viewModel;

import android.media.MediaPlayer;
import android.os.CountDownTimer;

import com.dynamic.dobler.elephantmath.database.entity.Ranking;
import com.dynamic.dobler.elephantmath.database.entity.RankingItem;
import com.google.firebase.database.DatabaseReference;

import java.util.List;

import androidx.lifecycle.ViewModel;

public class ChallengeViewModel extends ViewModel {


    public Ranking mRanking;
    public RankingItem mRankingItem;

    public List<RankingItem> mRankingItems;

    public MediaPlayer mp;
    public DatabaseReference mFirebaseDatabaseReference;

    public int timerMs = 10000;
    public int progressbarCount = 0;
    public int mLives = 3;
    public int mLevel = 2;
    public int mPoints = 0;
    public int mNumber1;
    public int mNumber2;
    public int mResult;
    public CountDownTimer mCountDownTimer;
    public boolean saved = false;
    public boolean mStarTed = false;



}

File: app/src/main/java/com/dynamic/dobler/elephantmath/ui/ChallengeViewModel.kt
package com.dynamic.dobler.elephantmath.ui

import android.media.MediaPlayer
import android.os.CountDownTimer
import androidx.lifecycle.ViewModel
import com.dynamic.dobler.elephantmath.database.entity.Ranking
import com.dynamic.dobler.elephantmath.database.entity.RankingItem
import com.google.firebase.database.DatabaseReference

class ChallengeViewModel : ViewModel() {
    var mRanking: Ranking? = null
    var mRankingItem: RankingItem? = null
    var mRankingItems: ArrayList<RankingItem>? = null
    var mp: MediaPlayer? = null
    var mFirebaseDatabaseReference: DatabaseReference? = null
    var timerMs = 10000
    var progressbarCount = 0
    var mLives = 3
    var mLevel = 2
    var mPoints = 0
    var mNumber1 = 0
    var mNumber2 = 0
    var mResult = 0
    var mCountDownTimer: CountDownTimer? = null
    var saved = false
    var mStarTed = false
}